tests/extract/test_extract_pipe.py

```diff
@@ -674,6 +699,33 @@ def raise_gen(item: int):
     assert_pipes_closed(raise_gen, long_gen)
 
 
+def test_close_on_async_generator() -> None:
+    global close_pipe_got_exit, close_pipe_yielding
+    close_pipe_got_exit = False
+    close_pipe_yielding = False
+
+    async def long_gen():
+        global close_pipe_got_exit, close_pipe_yielding
+
+        # will be closed by PipeIterator
+        try:
+            close_pipe_yielding = True
+            for i in range(0, 10000):
+                await asyncio.sleep(0.01)
+                yield i
+            close_pipe_yielding = False
+        # we have a different exception here
+        except asyncio.CancelledError:
```

Comment:
i expected to also get a generatorexit here, but it seems to not be the case.

Reply:
@z3z1ma right! but we wrap async generator in regular generator so I expect GeneratorExit here. if we get both it means that we are leaking some edge cases ie. when pool is closed and we cancel async generators using asyncio module. I'll investigate in separate branch